### PR TITLE
(WIP) Make Capifony compatible with symfony < 1.2

### PR DESCRIPTION
**Pull request by welcoMattic**

### CODE REVIEW

File: lib/capifony_symfony1.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def guess_symfony_orm
 def guess_symfony_lib
   symfony_version = capture("cd #{latest_release} && #{php_bin} ./symfony -V")
 
-  /\((.*)\)/.match(symfony_version)[1]
+  /1\.[0-1]\.[0-9]{2}/.match(symfony_version)[1]
```
Comment (willdurand) [inline lib/capifony_symfony1.rb]: mmh, it should work as is already;

File: lib/capifony_symfony1.rb [hunk 2]
```diff
@@ -149,7 +149,6 @@ def generate_sql_command(cmd_type, config)
     symfony.orm.build_classes             # 2. (Re)build the model
   end
   symfony.cc                              # 3. Clear cache
-  symfony.plugin.publish_assets           # 4. Publish plugin assets
```
Comment (willdurand) [inline lib/capifony_symfony1.rb]: why?

**Comment by willdurand** [comment]
ping @xophm1

File: lib/capifony_symfony1.rb (hunk 2 above)
Comment (xophm1) [inline lib/capifony_symfony1.rb]: Tested with symfony 1.4. No surprise: without this line, plugin assets are not available via symlink in the web dir.

File: lib/symfony1/symfony.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
   desc "Clears the cache"
   task :cc, :roles => :app, :except => { :no_release => true } do
-    run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony cache:clear'"
+    run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony clear-cache'"
```
Comment (xophm1) [inline lib/symfony1/symfony.rb]: This breaks compatibility with symfony 1.4:  Task "clear-cache" is not defined.

File: lib/symfony1/symfony.rb [hunk 4]
```diff
@@ -30,53 +30,46 @@
...
+  #     # restore logger level
+  #     self.logger.level = current_logger_level
+  #   end
+  # end
 
   namespace :project do
     desc "Disables an application in a given environment"
     task :disable, :roles => :app, :except => { :no_release => true } do
-      run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony project:disable #{symfony_env_prod}'"
+      run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony disable #{symfony_env_prod}'"
```
Comment (xophm1) [inline lib/symfony1/symfony.rb]: This breaks compatibility with symfony 1.4:  Task "disable" is not defined.

File: lib/symfony1/symfony.rb [hunk 5]
```diff
@@ -30,53 +30,46 @@
...
     task :permissions, :roles => :app, :except => { :no_release => true } do
-      run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony project:permissions'"
-    end
-
-    desc "Optimizes a project for better performance"
-    task :optimize do
-      prompt_with_default(:application, "frontend")
-
-      run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony project:optimize #{application}'"
+      run "#{try_sudo} sh -c 'cd #{latest_release} && #{php_bin} ./symfony fix-perms'"
```
Comment (xophm1) [inline lib/symfony1/symfony.rb]: Symfony 1.4: Task "fix-perms" is not defined.

**Comment by willdurand** [comment]
That's what I thought. This PR can't be merged, I don't think we can support symfony < 1.2...

File: lib/capifony_symfony1.rb (hunk 1 above)
Comment (welcoMattic) [inline lib/capifony_symfony1.rb]: @willdurand `php ./symfony -V` output (on 1.0.22) is : `symfony: version 1.0.22`

**Comment by welcoMattic** [comment]
@willdurand @xophm1 I'm working on it. The goal is to detect asap the Sf version to write task conditionally. Thanks for feedback

**Pull request closed without merging**